simphony/cuds/abstractlattice.py

```diff
@@ -22,6 +23,50 @@ class ABCLattice(object):
     """
     __metaclass__ = ABCMeta
 
+    def __init__(self, size, origin, a, b, c, alpha, beta, gamma):
+        """
+        Sets lattice constants a,b,c and angles between primitive vectors
+        """
+        self._size = size
+        self._origin = origin
+        self._a = a
+        self._b = b
+        self._c = c
+        self._alpha = alpha
+        self._beta = beta
+        self._gamma = gamma
+        self._set_primitive_vectors()
+        return self
+
+    def _set_primitive_vectors(self):
```

Comment:
I would rename this function to `estimate_primitive_vectors` and make it a module level function instead of a member function